core/src/Cabana_Core.hpp

```diff
@@ -56,6 +56,7 @@ bool cabana_initialized_kokkos = false;
 //---------------------------------------------------------------------------//
 // Initialize Kokkos, if it needs initialization.
 template <typename... Args>
+__attribute__ ((deprecated))
```

Comment:
Not sure about that one.  This is a GCC thing so it might not be portable across all compilers we want to support.  We might need to define yet another macro...

Reply:
I recommend for know you define an empty macro `#define CABANA_DEPRECATED` and you decorate deprecated stuff with it.

Reply:
There has been some talk about requiring C++14.  It would make our lives easier here because we would then just use the `[[deprecated]]` attribute.

Reply:
Of course I missed your comment in the description :)
Anyway IMO you should not use `__attribute__ ((deprecated))`

Reply:
So for now I could mark the code with the empty macro, without a warning?

Reply:
Yeah that's not a bad idea - if we move to C++14 soon then we can replace the empty macros with the C++14 attribute as @dalg24 indicated

Reply:
Or just wait until the C++14 decision